docs/kubernetes/operator/instrumenting-applications.md

```diff
@@ -0,0 +1,159 @@
+
+```yaml
+  exporter:
+    endpoint: http://opentelemetry-kube-stack-daemon-collector.opentelemetry-operator-system.svc.cluster.local:4318
+```
+
+- The **images** for the different languages: Used by the Operator to inject the right libraries into the Pods.
+
+```yaml
+  dotnet:
+    image: docker.elastic.co/observability/elastic-otel-dotnet:edge
+  java:
+    image: docker.elastic.co/observability/elastic-otel-javaagent:1.0.0
+  nodejs:
+    image: docker.elastic.co/observability/elastic-otel-node:0.4.1
+  python:
+    image: docker.elastic.co/observability/elastic-otel-python:0.3.0
+```
+
+## Configuring auto-instrumentation
+
+To enable auto-instrumentation, add the corresponding annotation to the pods of existing deployments (`spec.template.metadata.annotations`), or to the desired namespace (to auto-instrument all pods in the namespace):
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: myapp
+spec:
+  ...
+  template:
+    metadata:
+      annotations:
+        instrumentation.opentelemetry.io/inject-<LANGUAGE>: "opentelemetry-operator-system/elastic-instrumentation"
+      ...
+    spec:
+      containers:
+      - image: myapplication-image
+        name: app
+      ...        
+```
+
+where `<LANGUAGE>` is one of: `go` , `java`, `nodejs`, `python`, `dotnet`
+
+> [!NOTE]
+> Ensure you add the annotations at Pod level and not directly at the workload `spec` level (Deployment, Job, etc.)
+> After adding annotations to Pods or Namespaces, the applications must be restarted for the instrumentation injection to take effect.
+
+If you followed the proposed installation of the Operator using the provided `values.yaml`, the previous value (`"opentelemetry-operator-system/elastic-instrumentation"`) should be correct for the environment.
+
+In case you have multiple Instrumentation objects with different settings or images, ensure you point your pods to the the desired `Instrumentation` objects in the annotations.
+
+The possible values for the annotation are documented in the [Operator documentation](https://opentelemetry.io/docs/kubernetes/operator/automatic/#add-annotations-to-existing-deployments). For reference purposes, the values are:
+
+- `"true"`: to inject Instrumentation resource with default name from the current namespace.
+- `"my-instrumentation"`: to inject Instrumentation CR instance with name `"my-instrumentation"` in the current namespace.
+- `"my-other-namespace/my-instrumentation"`: to inject Instrumentation CR instance with name `"my-instrumentation"` from another namespace `"my-other-namespace"`.
+- `"false"`: do not inject.
+
+### Namespace based annotations example
+
+The following example creates a namespace with an annotation to instrument all pods of the namespace with `java` libraries.
+
+```
+kubectl create namespace java-apps
+
+#Annotate app namespace
+kubectl annotate namespace java-apps instrumentation.opentelemetry.io/inject-java="opentelemetry-operator-system/elastic-instrumentation"
```

Comment:
Can we use this example above too?

`kubectl annotate deployment <deployment-name> -n <namespace> instrumentation.opentelemetry.io/inject-java="opentelemetry-operator-system/elastic-instrumentation"`

Reply:
I believe `kubectl annotate deployment` won't work as that will add the annotations at Deployment level (`spec.metadata.annotations`) and not at Pod level (`spec.template.metadata.annotations`).

If there's a way to annotate the *Pods* belonging to a Deployment with `kubectl annotate` command we could definitely add it, but I believe there's no way.

And `kubectl annotate pod` is not an option because the pod will need a restart for auto-instrumentation to take place, and the restart (delete) make the annotation to go away.

Reply:
Can you sync with @jackshirazi and @gizas to confirm which commands for annotating a deployment and namespace will work?